localstack-core/localstack/services/events/archive.py

```diff
@@ -50,10 +64,10 @@ def __init__(
         self.set_state(ArchiveState.CREATING)
         self.set_creation_time()
         self.client: BaseClient = self._initialize_client()
-        self.event_bus_name = extract_event_bus_name(event_source_arn)
+        self.event_bus_name: EventBusName = extract_event_bus_name(event_source_arn)
```

Comment:
style: I would avoid the repetition of the typings within the constructor for the assignment because it's already at class level, but it doesn't harm either.